semgrep-core/src/analyzing/Constant_propagation.ml

```diff
@@ -116,15 +116,19 @@ let var_stats prog : var_stats =
 
   let hooks =
     { V.default_visitor with
-      V.kdef = (fun (k, _v) x ->
+      V.kdef = (fun (k, v) x ->
         match x with
         | { name = EN (Id (id,
                            { id_resolved = {contents = Some(_kind, sid)}; _})); _},
-          VarDef ({ vinit = Some _; _ }) ->
+          VarDef ({ vinit = Some e; _ }) ->
             let var = (H.str_of_ident id, sid) in
             let stat = get_stat_or_create var h in
             incr stat.lvalue;
-            k x
+            (* We can't do `k x` otherwise the variable-declaration itself is
+             * re-visited but now interpreted as an assignment, and we would be
```

Comment:
Please, add "because of v_vardef_as_assign_expr in Visitor_AST.ml".
This v_vardef_as_assign_expr is maybe not a good thing ... We should maybe do that in Generic_vs_generic
instead of in the visitor.